notion_client/client.py

```diff
@@ -102,11 +107,19 @@ def _build_request(
         path: str,
         query: Optional[Dict[Any, Any]] = None,
         body: Optional[Dict[Any, Any]] = None,
-        auth: Optional[str] = None,
+        auth: Optional[Union[str, Dict[str, str]]] = None,
```

Comment:
Let's use a [TypedDict](https://peps.python.org/pep-0589/) :P 

(and drop support for Python 3.7 since it has been implemented in 3.8)